### ISSUE TITLE
Keep should return results of transform

### ISSUE THREAD

**Issue by greghart** [cognitect-labs/transducers-js#31]
I was reading through the documentation, and noticed that `keep` was basically the same as `filter`. Looking into it a bit more, I believe `keep` is supposed to return the *results* of the transform that are not null as opposed to the original values, as laid out at https://clojuredocs.org/clojure.core/keep .

```javascript
var t = transducers;
var xf = t.keep(function(x) { if(typeof x == "string") return "cool"; });
t.into([], xf, [0,1,"foo",3,4,"bar"]); // ["foo","bar"] -> should return ['cool', 'cool']
```

**Comment by shaunc** [comment]
(Is this project still active?... no activity despite PRs waiting...)

**Comment by puredanger** [comment]
I agree with your analysis here - `keep` should be returning the results of the transform.

This project is still active, but is maintained by people with many other projects, so it only gets periodic attention. 

Also, as noted in the readme, this project does not accept pull requests.
